jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/SyncServerlessAppAction.kt

```diff
@@ -73,60 +98,135 @@ class SyncServerlessAppAction(private val codeOnly: Boolean = false) : AnAction(
                 return@thenAccept
             }
 
-            val warningSettings = SamDisplayDevModeWarningSettings.getInstance()
-            runInEdt {
-                if (warningSettings.showDevModeWarning) {
-                    if (!SyncServerlessAppWarningDialog(project).showAndGet()) {
-                        return@runInEdt
+            val templateFunctions = SamTemplateUtils.findFunctionsFromTemplate(project, templateFile)
+            val hasImageFunctions: Boolean = templateFunctions.any { (it as? SamFunction)?.packageType() == PackageType.IMAGE }
+            val lambdaType = if (hasImageFunctions) LambdaPackageType.Image else LambdaPackageType.Zip
+            val syncedResourceType = if (codeOnly) SyncedResources.CodeOnly else SyncedResources.AllResources
+
+            ProgressManager.getInstance().run(
+                object : Task.Backgroundable(project, "Fetching stacks", false) {
+                    var activeStacks = emptyList<StackSummary>()
+                    override fun run(indicator: ProgressIndicator) {
+                        activeStacks = project.getResourceNow(CloudFormationResources.ACTIVE_STACKS, forceFetch = true, useStale = false)
+                        activeStacks.size // no-op to ensure previous command is completed
```

Comment:
not sure why this would do anything

Reply:
Removed this, added Task.WithResult instead